File: 55-跳跃游戏.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int lens=nums.size(),maxPos=0, end=0;
        if(lens==1)
            return true;
        for(int i=0;i<lens-1;i++)
        {
            if(maxPos>=i)//表示能到达i
            {
                maxPos=max(maxPos,i+nums[i]);
                if(maxPos>=lens-1)
                    return true;
            }
        }
        return false;
    }
};

File: 55-跳跃游戏.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        lens = len(nums)
        if lens == 1:
            return True
        maxPos = nums[0]
        for i in range(1,lens):
            if maxPos >= i:
                maxPos = max(maxPos, nums[i]+i)
                if maxPos >= lens-1:
                    return True
            else:
                return False
        return False
